cmd/genji/shell/command.go

```diff
@@ -136,3 +141,193 @@ func displaySuggestions(in string) error {
+	ti, err := t.Info()
+	if err != nil {
+		return err
+	}
+
+	fields := ti.FieldConstraints
+	if len(fields) == 0 {
+		fmt.Printf("CREATE TABLE %s;\n", table)
+	} else {
+		fmt.Printf("CREATE TABLE %s(", table)
+		for i, f := range fields {
+			fmt.Printf("%s %s", fields[i].Path, strings.ToUpper(fields[i].Type.String()))
+			if f.IsPrimaryKey {
+				fmt.Printf(" PRIMARY KEY")
+			}
+
+			if f.IsNotNull {
+				fmt.Printf(" IS NOT NULL")
+			}
+
+			// Don't print the last comma.
+			if i == (len(fields) - 1) {
+				fmt.Println()
+				break
+			}
+
+			fmt.Println(",")
+		}
+
+		fmt.Println(");")
+	}
+
+	// Display indexes section.
+	indexes, err := t.Indexes()
+	if err != nil {
+		return err
+	}
+
+	for _, index := range indexes {
+		info := fmt.Sprintf("CREATE INDEX %s ON %s(%s);", index.Opts.IndexName, index.Opts.TableName, index.Opts.Path)
+		if index.Unique {
+			info = fmt.Sprintf("CREATE UNIQUE INDEX %s ON %s(%s);", index.Opts.IndexName, index.Opts.TableName, index.Opts.Path)
+		}
+
+		fmt.Println(info)
+	}
+
+	// Display inserts section.
+	return db.View(func(tx *genji.Tx) error {
+		q := fmt.Sprintf("SELECT * FROM %s", table)
+		res, err := tx.Query(q)
+		if err != nil {
+			return err
+		}
+		defer res.Close()
+
+		c, err := res.Count()
+		if err != nil {
+			return err
+		}
+		if c == 0 {
+			return nil
+		}
+
+		return res.Iterate(func(d document.Document) error {
+			var buf bytes.Buffer
+			err := document.ToJSON(&buf, d)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("INSERT INTO %s VALUES %s;\n", table, buf.String())
+			return nil
+		})
+
+	})
+}
+
+func getAllTables(db *genji.DB) ([]string, error) {
+	var tables []string
+	err := db.View(func(tx *genji.Tx) error {
+		res, err := tx.Query("SELECT table_name FROM __genji_tables")
+		if err != nil {
+			return err
+		}
+		defer res.Close()
+
+		return res.Iterate(func(d document.Document) error {
+			var tableName string
+			err = document.Scan(d, &tableName)
+			if err != nil {
+				return err
+			}
+			tables = append(tables, tableName)
+			return nil
+		})
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(tables) == 0 {
+		return nil, errors.New("nothing to dump")
+	}
+
+	return tables, nil
+}
+
+func runDumpCmd(db *genji.DB, table string) error {
+	switch table {
+	case "":
+		// Get all tables and ensure there is at least one table.
+		tables, err := getAllTables(db)
+		if err != nil {
+			return err
+		}
+
+		dtx, err := getDatabaseTransaction(db)
+		if err != nil {
+			return err
+		}
+		defer dtx.Commit()
+		defer fmt.Println("COMMIT;")
+
+		for _, t := range tables {
+			err = dumpTable(db, dtx, t)
+			if err != nil {
+				return err
+			}
+			fmt.Println()
+		}
+
+		break
+	default:
+		// Insure the given table exists.
+		err := db.View(func(tx *genji.Tx) error {
+			_, err := tx.GetTable(table)
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+
+		dtx, err := getDatabaseTransaction(db)
```

Comment:
We don't need to create a transaction and commit it, we only need to display that statement. You can remove `getDatabaseTransaction` and `Commit` as it has no effect.